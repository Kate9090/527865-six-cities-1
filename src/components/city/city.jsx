import React, {Component} from "react";
import PropTypes from 'prop-types';


import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer';

class City extends Component {
  constructor(props) {
    super(props);

    this.handleUserChoose = this.handleUserChoose.bind(this);
  }

  handleUserChoose(city, num, evt) {
    evt.preventDefault();
    this.props.onUserAnswer(city, num);
    this.setActiveItem();
  }

  render() {
    const {cityObject, idx} = this.props;

    return <li className="locations__item">
      <a onClick={(e) => this.handleUserChoose(cityObject.city, idx, e)}
        // onMouseLeave={() => this.setUnActiveItem()}
        className={`locations__item-link tabs__item`}
        //  ${idx === current ?
        // `tabs__item--current` : ``}`}
        href="#">
        <span>{cityObject.city}</span>
      </a>
    </li>;
  }
}

City.propTypes = {
  onUserAnswer: PropTypes.func,
  cityObject: PropTypes.shape({
    offerCoord: PropTypes.array.isRequired,
    city: PropTypes.string.isRequired,
  }).isRequired,
  idx: PropTypes.number.isRequired,
};

export {City};

const mapDispatchToProps = (dispatch) => ({
  onUserAnswer: (city, num) => {
    dispatch(ActionCreator[`NEW_CITY`](city, num));
  },
});

export default connect(
    null,
    mapDispatchToProps
)(City);
