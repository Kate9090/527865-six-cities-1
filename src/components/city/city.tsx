import * as React from 'react';

import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer/user/user';

interface Props {
  onUserAnswer: (city: string, num: number, cities: string[]) => void,
  city: string,
  idx: number,
  onCardClick: (city: string) => void,
  current: string,
  cities: string[],
};

const City: React.FunctionComponent<Props> = (props) => {

  const _handleUserChoose = (city, num, cities, evt) => {
    evt.preventDefault();
    props.onUserAnswer(city, num, cities);
    props.onCardClick(city);
  };

  const {city, idx, cities} = props;

  return <li className="locations__item">
    <a onClick={(e) => _handleUserChoose(city, idx, cities, e)}
      className={props.current === city ? `locations__item-link tabs__item tabs__item--active` : `locations__item-link tabs__item`}
      href="#">
      <span>{city}</span>
    </a>
  </li>;
};

export {City};

const mapDispatchToProps = (dispatch) => ({
  onUserAnswer: (city, num, cities) => {
    dispatch(ActionCreator.selectCity(city, num, cities));
  },
});

export default connect(
    null,
    mapDispatchToProps
)(City);
