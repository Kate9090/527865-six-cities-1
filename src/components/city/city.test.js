import React from "react";
import renderer from 'react-test-renderer';

import {City} from './city.jsx';

import offerCities from '../../mocks/offers-city';

const mock = offerCities[2];

it(`renders correctly City List`, () => {
  const tree = renderer
    .create(<City
      cityObject = {mock}
      idx = {2}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
