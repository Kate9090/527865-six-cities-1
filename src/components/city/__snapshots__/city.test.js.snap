// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly City List 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item"
    href="#"
    onClick={[Function]}
  >
    <span>
      Moscow
    </span>
  </a>
</li>
`;
